docs/proposals/05-versioning-policy.md

```diff
@@ -7,26 +7,21 @@
+- **preview:** After a fresh release of a new Kubernetes **minor** version (e.g. v1.17.0) the operator could tag it as _preview_ until he has gained sufficient experience. It will not become the default in the Gardener Dashboard until he promotes that minor version to _supported_, which could happen a few weeks later with the first patch version.
 
-Example: Assuming the Kubernetes community maintains at present the following minor versions v1.16.1, v1.15.4 and v1.14.9, then we classify as follows:
+- **supported:** The operator would tag the latest Kubernetes patch versions of the actual (if not still in _preview_) and the last three minor Kubernetes versions as _supported_ (e.g. v1.16.1, v1.15.4, v1.14.9 and v1.13.12). The latest of these becomes the default in the Gardener Dashboard (e.g. v1.16.1).
 
-- **preview:** After a fresh release of a new Kubernetes **minor** version (e.g. v1.17.0) we tag it as _preview_ until we have gained sufficient experience. It will not become the default in the Gardener Dashboard until we promote that minor version to _supported_, which usually happens a few weeks later with the first patch version (as a rule of thumb, e.g. v1.17.1, but it could be also v1.17.0 if it was stable and there was no patch version or v1.17.2 or higher if there were many because quality was not good enough in the beginning of that Kubernetes minor release).
+- **deprecated:** The operator could decide, that he generally wants every version that is not the latest patch version as _deprecated_ and flag this versions accordingly (e.g. v1.16.0 and older, v1.15.3 and older, 1.14.8 and older as well as v1.13.11 and older). He could also tag all versions (latest or not) of every Kubernetes minor release that is neither the actual nor one of the last three minor Kubernetes versions as _deprecated_, too (e.g. v1.12.x and older). Deprecated versions will eventually expire (i.e. removed).
 
-- **supported:** We tag the latest Kubernetes patch versions of the actual (if not still in _preview_) and the last two minor Kubernetes versions as _supported_ (e.g. v1.16.1, v1.15.4 and v1.14.9). The latest of these becomes the default in the Gardener Dashboard (e.g. v1.16.1).
+- **expired:** A Kubernetes version will be flagged as expired, when its `expirationDate` as per the `cloudprofile` is reached. After that date has passed, users cannot create new clusters with that version anymore and any cluster that is on that version will be forcefully migrated in its next maintenance time window, even if the owner has opted out of automatic cluster updates! **The forceful update will pick the latest patch version of the current minor Kubernetes version. If that's expired as well, the update process repeats until a non-expired Kubernetes version is reached, which could also result in an update of a minor version!**
```

Comment:
I'm not sure whether we need an own class for this. Versions whose `expirationDate` is expired are automatically in this class, and I don't see the need that an operator changes this manually in the cloud profile. Another approach could be that the cloudprofile controller automatically updates the class to `expired` once the `expirationDate` expires, however, this will require quite some implementation effort. What do we want? cc @vlerenc

Reply:
I guess it's fine to do nothing and rephrase this paragraph, so that it becomes clear that logically that's a different state only, i.e. any version (should be deprecated, though) whose expiration date has lapsed is automatically/logically considered expired. Fine with me.

Reply:
added suggestions. @rfranzke please review